pkg/collector/corechecks/cluster/kubernetesapiserver/kubernetes_apiserver.go

```diff
@@ -174,7 +181,7 @@ func (k *KubeASCheck) Configure(senderManager sender.SenderManager, _ uint64, co
 	}
 
 	if k.instance.UnbundleEvents {
-		k.eventCollection.Transformer = newUnbundledTransformer(clusterName, tagger.GetTaggerInstance(), k.instance.CollectedEventTypes)
+		k.eventCollection.Transformer = newUnbundledTransformer(clusterName, tagger.GetTaggerInstance(), k.instance.CollectedEventTypes, k.instance.BundleUnspecifedEvents)
 	} else {
 		k.eventCollection.Filter = convertFilters(k.instance.FilteredEventTypes)
```

Comment:
What if someone has:
```
unbundle_events: true
bundle_unspecifed_events: true
filtered_event_types: ["reason!=FailedGetScale","involvedObject.kind==Pod","type==Normal"]
```

The change as it currently stands provides people the option to:
- bundle everything, but exclude what they want to exclude
- exclude everything, but unbundle only what they want to keep
- unbundle specific things, but bundle everything else

There is no possibility for users of this config to filter out events they don't want showing in the platform at all. I wonder if we could pull this line out from this if/else block (I don't know if doing so will have unintended consequences when `unbundle_events: true` but `bundle_unspecifed_events: false`), or otherwise abstract the creation of this filter to expose it only in the cases where a bundler will be used at any point

Reply:
https://github.com/DataDog/datadog-agent/pull/25642/commits/480b18d547a783ac202655c2b2758bf788dd85ce addressed your suggestion and I updated "Describe how to test/QA your changes" to show you the result.

Reply:
https://github.com/DataDog/datadog-agent/pull/25642/commits/ccd892f3a29240510bd70685b91fb6c575ecb770 addressed your suggestion and I updated "Describe how to test/QA your changes" to show you the result.